MAINTAINERS: add AndesTech Platforms section

jimmyzhe opened the pull request:
The AndesTech board/soc/dts files are currently orphaned and Andes group is willing to maintain the Andes platform.

Migrate Andes related paths from CODEOWNERS to MAINTAINERS.yml and add myself as a maintainer and @kevinwang821020, @wtlee1995 as collaborators.
Remove @cwshu since he is no longer maintaining the Andes platform.

jimmyzhe committed:
MAINTAINERS: add AndesTech Platforms

Add AndesTech Platforms section and add me jimmyzhe, kevinwang821020
as maintainer and collaborator.

Signed-off-by: Jimmy Zheng <jimmyzhe@andestech.com>

jimmyzhe commented:
Hi @nashif, @stephanosio , could you help review this change? 

There are some open PRs related to the Andes platform (#76211, #78521), and I hope the Andes platform can be maintained.

nashif commented on MAINTAINERS.yml:
```diff
@@ -261,6 +261,24 @@ Ambiq Platforms:
+AndesTech Platforms:
+  status: maintained
+  maintainers:
+    - jimmyzhe
```
you need to follow the process here https://docs.zephyrproject.org/latest/project/project_roles.html and be added to the project first as contributor before this happens

jimmyzhe replied:
Got it, thank you for your guidance.

Hi @kevinwang821020, @wtlee1995, please also nominate yourselves and join the project.

jimmyzhe replied:
Hi @nashif,
Could you help review this again? 
If I missed any step, please let me know, thank you.

jimmyzhe commented:
Rebased and split updates to MAINTAINERS.yml into a standalone commit.

@kevinwang821020, @wtlee1995, and I mainly work on the Andes platform. The related PRs can be found in the following nomination issues:
- #78760
- #78824
- #78825

This PR adds a new area, "AndesTech Platforms" and promotes me @jimmyzhe, as a new maintainer. I think this PR requires TSC review, and I would appreciate your review.

jimmyzhe commented:
Reopening because #88530.

@kevinwang821020 and I are willing to maintain the Andes platform on upstream.
It would be helpful if Andes-related issues/PRs could be assigned.
